internal/client/graph_api.go

```diff
@@ -49,16 +61,29 @@ func NewGraphAPI(options GraphAPIOptions) *GraphAPI {
 	}
 }
 
+// In order for all importers ran at the same time to not read again at the same time we want to add a random duration to the
+// anti entropy duration of 1/4 of the duration.
+func (gapi *GraphAPI) isAntiEntropyDurationExceeded() bool {
+	delta := int64(gapi.options.AntiEntropyDuration) / 4
+	max := delta
+	min := -delta
+	randDelta := time.Duration(rand.Int63n(max-min+1) + min)
+	return gapi.currentGraphLastServerSync.
+		Add(gapi.options.AntiEntropyDuration + randDelta).
+		Before(time.Now())
+}
+
 // CreateTransaction create a full graph transaction. This kind of transaction will diff the new graph
 // with previous version of it.
 func (gapi *GraphAPI) CreateTransaction() (*Transaction, error) {
-	if gapi.currentGraph == nil {
+	if gapi.currentGraph == nil || gapi.isAntiEntropyDurationExceeded() {
```

Comment:
This is not a major issue, but this is evaluated at each request, which creates a bias towards lower values of the AE rand range.
You may want to replace `currentGraphLastServerSync` by something like `currentGraphStaleAfter` and compute the rand value once.

Reply:
Yes, I knew that. I though it would be ok to randomize even further but I'm fine with the other approach too. I'll do that.

Reply:
done

Reply:
Recomputing the random value each time reduces the randomness here.